### PR TITLE
Cosmos: CFP and doc related changes

### PR DESCRIPTION
**Pull request by milismsft**
- Update Change Feed Processor public APIs and documentation.
- Add implementation and respective test case for estimating the Change Feed Processor lag between the last processed batch of feeds (check-pointed in the respective lease document) and the latest feed document (see https://github.com/Azure/azure-sdk-for-java/issues/5744).
- Remove container readAllItems() from public API surface.
- Add new APIs for createContainerIfNotExists() that will accept a throughput properties object.
- Fix docs; finish sentences with ".", rewording.
- Add createDatabaseIfNotExist() method that takes a throughput properties object.
- Remove readProvisionedThroughput() and replaceProvisionedThroughput(); with the introduction of auto-scale features the throughput must be explicitly handled via ThroughputProperties class.
 - Remove/refactor public APIs that take throughput as an integer to use a full throughput properties object.

This PR handles following issues : 
* https://github.com/Azure/azure-sdk-for-java/issues/11191
* https://github.com/Azure/azure-sdk-for-java/issues/11219
* https://github.com/Azure/azure-sdk-for-java/issues/11190

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosContainer.java [hunk 1]
```diff
@@ -46,134 +46,124 @@
...
     }
 
     /**
-     * Read provisioned throughput integer.
+     * Read provisioned throughput for the current container.
      *
-     * @return the integer. null response indicates database doesn't have any provisioned RUs
+     * @return provisioned throughput for the current container or null response if doesn't have any provisioned RUs.
      */
     public Integer readProvisionedThroughput() {
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosContainer.java:122]: We should hide / remove this API - as discussed in this ticket : https://github.com/Azure/azure-sdk-for-java/issues/11191

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosDatabase.java [hunk 2]
```diff
@@ -458,7 +497,7 @@ private CosmosUserResponse convertUserResponse(CosmosAsyncUserResponse response)
     /**
      * Read provisioned throughput integer.
      *
-     * @return the integer. null response indicates database doesn't have any provisioned RUs
+     * @return the integer. null response indicates database doesn't have any provisioned RUs.
      */
     public Integer readProvisionedThroughput() {
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosDatabase.java:502]: Same here - hide / remove this API - https://github.com/Azure/azure-sdk-for-java/issues/11191

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosDatabase.java [hunk 3]
```diff
@@ -467,8 +506,8 @@ public Integer readProvisionedThroughput() {
     /**
      * Replace provisioned throughput integer.
      *
-     * @param requestUnitsPerSecond the request units per second
-     * @return the integer
+     * @param requestUnitsPerSecond the request units per second.
+     * @return the integer.
      */
     public Integer replaceProvisionedThroughput(int requestUnitsPerSecond) {
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosDatabase.java:512]: Same here : hide / remove this API - https://github.com/Azure/azure-sdk-for-java/issues/11191

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosContainer.java (hunk 1 above)
Comment (milismsft) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosContainer.java]: Fixed.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosDatabase.java (hunk 3 above)
Comment (milismsft) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosDatabase.java]: Fixed.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosDatabase.java (hunk 2 above)
Comment (milismsft) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosDatabase.java]: Fixed.

**Pull request merged**